agenta-backend/agenta_backend/migrations/mongo_to_postgres/utils.py

```diff
@@ -0,0 +1,155 @@
+import os
+import asyncio
+from datetime import datetime, timezone
+
+from pymongo import MongoClient
+from bson import ObjectId, DBRef
+from sqlalchemy import MetaData, Column, String, DateTime, text, create_engine
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+import uuid_utils.compat as uuid
+from sqlalchemy.future import select
+
+from agenta_backend.models.db_engine import db_engine
+
+from agenta_backend.models.db_models import (
+    IDsMappingDB,
+    Base,
+)
+
+BATCH_SIZE = 1000
+
+# MongoDB connection
+MONGO_URI = os.environ.get("MONGODB_URI")
+DATABASE_MODE = os.environ.get("DATABASE_MODE")
+mongo_client = MongoClient(MONGO_URI)
+mongo_db_name = f"agenta_{DATABASE_MODE}"
+mongo_db = mongo_client[mongo_db_name]
+
+migration_report = {}
+
+
+async def drop_all_tables():
+    """Drop all tables in the database."""
+    async with db_engine.engine.begin() as conn:
+        await conn.run_sync(Base.metadata.reflect)
+        # Drop all tables with CASCADE option
+        for table in reversed(Base.metadata.sorted_tables):
+            await conn.execute(text(f"DROP TABLE IF EXISTS {table.name} CASCADE"))
+
+
+async def create_all_tables(tables):
+    """Create all tables in the database."""
+    async with db_engine.engine.begin() as conn:
+        for table in tables:
+            print(f"====================== Creating table for {table.__name__}")
+            await conn.run_sync(table.metadata.create_all)
+    print("All tables dropped and created.")
+
+
+async def store_mapping(table_name, mongo_id, uuid):
+    """Store the mapping of MongoDB ObjectId to UUID in the mapping table."""
+    id = generate_uuid()
```

Comment:
[we should not use the name id in python](https://stackoverflow.com/questions/77552/id-is-a-bad-variable-name-in-python), it conflicts with an internal variable name. Instead use `id_`